File: demo4/src/main/java/com/infobip/demo3/service/UserService.java
package com.infobip.demo3.service;

import com.infobip.demo3.model.User;

import java.util.List;

public interface UserService {
    default User saveUser(User user) {
        int id = (int) Math.random();
        return saveUser(user);
    }
    List<User> getAllUsers();
    User getUserById(int id);
    String deleteUser(int id);
    User updateUser(User user);

    User findByEmail(String email);
    User findByNumber(String number);
    User findByUsername(String username);
}

File: sms-api-kotlin/userservice/src/main/kotlin/com/example/userservice/service/UserService.kt
package com.example.userservice.service

import com.example.userservice.model.User


interface UserService {
    fun saveUser(user: User): User {
        val id = Math.random().toLong()
        return saveUser(user)
    }

    val getAllUsers: List<User>

    fun getUserById(id: Long): User
    fun deleteUser(id: Long)
    fun updateUser(id: Long, user: User): User
    fun findByUsername(username: String): User
}


